machine/v1beta1/types_alibabaprovider.go

```diff
@@ -0,0 +1,320 @@
+	//Default value: default.
+	// +optional
+	Tenancy InstanceTenancy `json:"tenancy,omitempty"`
+
+	// UserDataSecret contains a local reference to a secret that contains the
+	// UserData to apply to the instance
+	// +optional
+	UserDataSecret *corev1.LocalObjectReference `json:"userDataSecret,omitempty"`
+
+	// CredentialsSecret is a reference to the secret with alibabacloud credentials. Otherwise, defaults to permissions
+	// provided by attached RAM role where the actuator is running.
+	// +optional
+	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
+
+	// Tags are the set of metadata to add to an instance.
+	// +optional
+	Tags []Tag `json:"tag,omitempty"`
+}
+
+// ResourceTagReference is a reference to a specific AlibabaCloud resource by ID, or tags.
+// Only one of ID or Tags may be specified. Specifying more than one will result in
+// a validation error.
+type AlibabaResourceReference struct {
+	// ID of resource
+	// +optional
+	ID string `json:"id,omitempty"`
+
+	// Tags is a set of metadata based upon ECS object tags used to identify a resource
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AlibabaCloudMachineProviderConfigList contains a list of AlibabaCloudMachineProviderConfig
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type AlibabaCloudMachineProviderConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []AlibabaCloudMachineProviderConfig `json:"items"`
+}
+
+// AlibabaCloudMachineProviderStatus is the Schema for the alibabacloudmachineproviderconfig API
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type AlibabaCloudMachineProviderStatus struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// InstanceID is the instance ID of the machine created in alibabacloud
+	// +optional
+	InstanceID *string `json:"instanceId,omitempty"`
+
+	// InstanceState is the state of the alibabacloud instance for this machine
+	// +optional
+	InstanceState *string `json:"instanceState,omitempty"`
+
+	// Conditions is a set of conditions associated with the Machine to indicate
+	// errors or other status
+	// +optional
+	Conditions []AlibabaCloudMachineProviderCondition `json:"conditions,omitempty"`
+}
+
+// AlibabaCloudMachineProviderCondition is a condition in a AlibabaCloudMachineProviderStatus
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type AlibabaCloudMachineProviderCondition struct {
```

Comment:
use metav1.Condition instead of rolling your own

Reply:
I'm not familiar with the reason this was done. I do see this is used in each of the other cloud provider's:
- https://github.com/openshift/api/blob/master/machine/v1beta1/types_awsprovider.go#L257
- https://github.com/openshift/api/blob/master/machine/v1beta1/types_azureprovider.go#L257
- https://github.com/openshift/api/blob/master/machine/v1beta1/types_gcpprovider.go#L206
- https://github.com/openshift/api/blob/master/machine/v1beta1/types_vsphereprovider.go#L118

I'm also not sure I understand the repercussion so moving to metav1.Condition. I'll see what happens if I change that in the cluster-api-provider-alibaba.

Maybe @JoelSpeed or @elmiko or @alexander-demichev can comment since this is their domain of expertise.

Reply:
With a few small fixes in the forms of casting `machinev1.ConditionType` to `string()` and removing the original `condition.LastProbeTime` I was able to compile the cluster-api-provider-alibaba.

I'm not as confident what this means so I'll need more input from @JoelSpeed and his team.  WDYT?

Reply:
I think this is ok to change if it works, I can't see much difference between the two. I don't have context for why we have separate conditions on other platforms, this predates me

Reply:
I will proceed by removing the conditions here. I think this is cleaner but does leave Alibaba at odds with the other providers. We might want to keep these in sync to reduce 1-offs. This would require some API refactoring for the other cloud providers.

Reply:
i think changing in favor of metav1.Condition seems ok for this part of the API. i am going to make a jira card for us to evaluate the other provider conditions as well. it seems like we could refactor all of these to use metav1.Condition, but due to the organic growth of the API we have not done it yet.